code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -411,9 +412,18 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
 
 /datum/gas_mixture/react(turf/open/dump_location)
 	. = NO_REACTION
+	if(volume == 0)
+		return
+	var/possible
+	for(var/gas in gases)
+		if(is_type_in_typecache(gas,nonreactive_gases))

```

Comment:
This less so, this should be computed from gas properties not a static list. And i'm not entirely convinced this will be in-plus in high-load scenario.

Reply:
do 
```dm
for(var/I in gases)
    var/datum/gas/gas = I
    if(nonreactive_gases[gas.type])
        continue
```

This is fine otherwise, see the rest of my review for why